Documentation/CodingGuidelines/CodingBestPractices/NamedArguments.rst

```diff
@@ -0,0 +1,247 @@
+..  include:: /Includes.rst.txt
+..  index:: pair: Coding guidelines; Named Arguments
+..  _cgl-named-arguments:
+
+===============
+Named Arguments
+===============
+
+`Named arguments <https://www.php.net/manual/en/functions.arguments.php#functions.named-arguments>`__,
+also known as “named parameters”, were `introduced in PHP 8 <https://wiki.php.net/rfc/named_params>`__,
+offering a new approach to passing arguments to functions. Instead of relying on
+the position of parameters, developers can now specify arguments based on their
+corresponding parameter names:
+
+..  code-block:: php
+    :caption: Named arguments example
+
+    <?php
+    function createUser($username, $email)
+    {
+        // code to create user
+    }
+    createUser(email: 'john.doe@example.com', username: 'john');
+
+This document discusses the use of named arguments within the TYPO3 core ecosystem,
+outlining best practices for TYPO3 extension and core developers regarding the
+adoption and avoidance of this language feature.
+
+
+Named Arguments in Public APIs
+==============================
+
+The key consideration when using this feature is outlined in the
+`PHP documentation <https://php.watch/versions/8.0/named-parameters>`__:
+
+    With named parameters, the name of the function/method parameters become part of
+    the public API, and changing the parameters of a function will be a semantic
+    versioning breaking-change. This is an undesired effect of named parameters feature.
```

Comment:
why? it has it's own docs. and people who don't know this basic should google or something?

Reply:
We try to crosslink terms so that the documentation can be accessible for any level of developer. But Chris and I can take care of crosslinking later :-)